src/components/Carousel.tsx

```diff
@@ -1,18 +1,105 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Carousel.scss';
+import cn from 'classnames';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+interface Props {
+  images: string[],
+  step: number,
+  frameSize: number,
+  itemWidth: number,
+  animationDuration: number,
+  infinite: boolean,
+}
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const gap = 20;
+  const totalGapValue = gap * (frameSize - 1);
+  const visibleImagesWidth = itemWidth * frameSize + totalGapValue;
+
+  const [translateValue, setTranslateValue] = useState(0);
+
+  const newTranslateValue = step * (itemWidth + gap);
+
+  const allImagesWidth = (10 * itemWidth) + (9 * gap);
+  const extremeTranslatePoint = allImagesWidth - visibleImagesWidth;
+
+  const handlePrev = () => {
+    let value = 0;
+
+    value = Math.min(translateValue + newTranslateValue, 0);
+
+    setTranslateValue(value);
+  };
```

Comment:
redundant to make variable value. change directly state in different cases and use prevState
also infinite should work for prev button
```suggestion
  const handlePrev = () => {
       setTranslateValue(
      translateValue + newTranslateValue >= 0
        ? 0
        : prevPos => prevPos + newTranslateValue,
    );

    if (infinite && translateValue + newTranslateValue === newTranslateValue) {
      setTranslateValue(extremeTranslatePoint);
    }
  };
```